### PR TITLE
Add support for `callable` inbuilt function

### PR DESCRIPTION
**Pull request by Smit-create**

### CODE REVIEW

File: tests/constants1.py [hunk 1]
```diff
@@ -39,3 +39,15 @@ def test_bool():
     a = bool('t')
     a = bool(2.3)
     assert a == True
+
+
+def test_callable():
+    a: bool
+    b: i32
+    b = 2
+    a = callable(test_len)
```
Comment (namannimmo10) [inline tests/constants1.py:48]: We could also test `callable` on other built-in functions:
```python
callable(print) == True
callable(len) == True
```

Comment (Smit-create) [reply tests/constants1.py:48]: Ohh, yeah thanks for pointing. One solution to that can be to manually store built-in functions' names in a vector and then check for them.

**Review by certik — APPROVED** [review]
I think this is fine.

File: tests/constants1.py (hunk 1 above)
Comment (certik) [reply tests/constants1.py:48]: The current check is fine.

Comment (certik) [reply tests/constants1.py:48]: We have to move all these built-in functions into `FunctionCall` with a compile time value. Then this check would work for most. Finally, there are some like "print", which would need to be handled in a different way --- probably the LPython frontend would know. There might be some corner cases that would need some design work, such as:
```
a = print
callable(a)
```
And passing `a` at runtime...

**Pull request merged**